pkg/cmd/audit/audit.go

```diff
@@ -154,6 +159,9 @@ func (o *AuditOptions) Run() error {
 	if o.failedOnly {
 		filters = append(filters, &FilterByFailures{})
 	}
+	if len(o.stages) > 0 {
```

Comment:
this is guaranteed to cut the list by half.  Take it to just below namespaces for efficiency when filtering.